frontend/src/components/metafilter.js

```diff
@@ -0,0 +1,222 @@
+  const [isValueOpen, setIsValueOpen] = useState(false);
+
+  const [valueOptions, setValueOptions] = useState([]);
+  const [valueSelections, setValueSelections] = useState([]);
+  const [fieldOptions, setFieldOptions] = useState([]);
+
+  const onFieldSelect = (_event, selection) => {
+    // clear value state too, otherwise the old selection remains selected but is no longer visible
+    setFieldSelection(selection);
+    setisFieldOpen(false);
+    setValueSelections([]);
+    setValueOptions([]);
+    setIsValueOpen(false);
+
+  };
+
+  const onValueSelect = (_event, selection) => {
+    // update state and call setFilter
+    let updatedValues = (valueSelections.includes(selection))
+      ? valueSelections.filter(item => item !== selection)
+      : [...valueSelections, selection];
+
+    setValueSelections(updatedValues);
+    setIsValueOpen(false);
+    if (updatedValues.length > 0) {
+      setFilter(id, fieldSelection, updatedValues.join(';'));
+    }
+
+    // after filter was applied, remove the values from the fields
+    setValueSelections([]);
+    setFieldSelection([]);
+  };
+
+  const onFieldClear = () => {
+    setFieldSelection([]);
+    setisFieldOpen(false);
+    setIsValueOpen(false);
+    setValueOptions([]);
+    setValueSelections([]);
+  };
+
+  const onValueClear = () => {
+    setValueSelections([]);
+    setIsValueOpen(false);
+    setFilter(id, fieldSelection, []);
+  };
+
+  useEffect(() => {
+    // fetch the available values for the selected meta field
+    if (fieldSelection.length !== 0) {
+      let api_filter = toAPIFilter(activeFilters).join();
+
+      let projectId = primaryObject ? primaryObject.id : '';
+
+      // make runId optional
+      let params = {
+        group_field: fieldSelection,
+        additional_filters: api_filter,
+        project: projectId
+      };
+      if (runId) {
+        params['run_id'] = runId;
+      } else {
+        params['days'] = 30;
+      }
+
+      HttpClient.get(
+        [Settings.serverUrl, 'widget', 'result-aggregator'],
+        params
+      )
+        .then(response => HttpClient.handleResponse(response))
+        .then(data => {
+          setValueOptions(data);
+        });
+    }
+  }, [fieldSelection, activeFilters, primaryObject, runId]);
+
+  useEffect(() => {
+    // Fetch field options for the project
+    HttpClient.get([Settings.serverUrl, 'project', 'filter-params', primaryObject.id])
+      .then(response => HttpClient.handleResponse(response))
+      .then(data => {
+        setFieldOptions(data);
+      });
+  }, [primaryObject.id]);
+
+  let values_available = valueOptions.length > 0;
+  let value_placeholder = 'Select a field first' ; // default instead of an else block
+  if (fieldSelection.length > 0 && values_available) {
+    value_placeholder = 'Select value(s)';
+  }
+  else if (fieldSelection.length > 0 && !values_available) {
+    value_placeholder = 'No values for selected field';
+  }
```

Comment:
nitpick we can replace it with:
```
  const valuePlaceholder = !fieldSelection.length
  ? 'Select a field first'
  : values_available
    ? 'Select value(s)'
    : 'No values for selected field';
```
also vars can be camelCase

Reply:
heh, I looked at this block for a moment and thought about golfing it but decided to ignore it :)